sysfs/clocksource.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !windows
+
+package sysfs
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+// Clocksource contains metrics related to the clock source
+type Clocksource struct {
+	Name      string
+	Available []string
+	Current   string
+}
+
+// NewClocksource returns clocksource information including current and available clocksources.
+func NewClocksource() ([]Clocksource, error) {
+	fs, err := NewFS(DefaultMountPoint)
+	if err != nil {
+		return []Clocksource{}, err
+	}
+
+	return fs.NewClocksource()
+}
+
+// NewClocksource returns clocksource information including current and available clocksources.
+func (fs FS) NewClocksource() ([]Clocksource, error) {
```

Comment:
This should probably be Clocksources(), following #44 
I'm not sure though about the top level function.. tricky.. But since the structs are mainly just holding data for for consistency with the interface on `FS` we should probably call both Clocksources() (or maybe ClockSources?)

Reply:
Based on the comments here and in issue #44, I removed the top level constructor function and changed the FS method name to `ClockSources()`.